gtdynamics/universal_robot/sdf.cpp

```diff
@@ -72,9 +72,31 @@ JointParams ParametersFromSdfJoint(const sdf::Joint &sdf_joint) {
   return parameters;
 }
 
+// Get Link posein base frame from sdf::Link object
+Pose3 GetBTLFromSdfLink(const sdf::Link &sdf_link) {
+// Call SemanticPose::Resolve so the pose is resolved to the correct frame
+  /// http://sdformat.org/tutorials?tut=pose_frame_semantics&ver=1.7&cat=specification&
+  // Get non-const pose of link in the frame of the joint it is connect to
+  // (http://wiki.ros.org/urdf/XML/link).
+  auto raw_pose = sdf_link.RawPose();
+
+  // Update from joint frame to base frame in-place.
+  // Base frame is denoted by "".
+  auto errors = sdf_link.SemanticPose().Resolve(raw_pose, "");
+  // If any errors in the resolution, throw an exception.
+  if (errors.size() > 0) {
+    throw std::runtime_error(errors[0].Message());
+  }
+  // Pose is updated from joint frame to base frame.
+  const auto bTl = Pose3FromIgnition(raw_pose);
+
+  return bTl;
+}
+
 Pose3 GetJointFrame(const sdf::Joint &sdf_joint,
-                    const LinkSharedPtr &parent_link,
-                    const LinkSharedPtr &child_link) {
+                    const LinkSharedPtr &parent_link, const sdf::Link &parent_sdf_link,
```

Comment:
Do we still need parent_a link and child_link?

Reply:
We don't. I fixed the function. Done